Source/WebCore/WebCoreSwift/CryptoKitShim.swift

```diff
@@ -48,47 +49,138 @@ public final class AesGcm: NSObject {
+            let symKey = SymmetricKey(
+                data: UnsafeBufferPointer(start: key, count: Int(keySize)))
             let sealedBox: AES.GCM.SealedBox
             if additionalDataSize > 0 {
                 sealedBox = try AES.GCM.seal(
                     message,
-                    using: symKey, 
+                    using: symKey,
                     nonce: nonce,
-                    authenticating: UnsafeBufferPointer(start: additionalData, count: Int(additionalDataSize))
+                    authenticating: UnsafeBufferPointer(
+                        start: additionalData, count: Int(additionalDataSize))
                 )
             } else {
                 sealedBox = try AES.GCM.seal(
                     message,
-                    using: symKey, 
+                    using: symKey,
                     nonce: nonce
                 )
             }
-            sealedBox.ciphertext.copyBytes(to: cipherText, count: Int(plainTextSize));
-            sealedBox.tag.copyBytes(to: cipherText + Int(plainTextSize), count: sealedBox.tag.count);
+            sealedBox.ciphertext.copyBytes(to: cipherText, count: Int(plainTextSize))
+            sealedBox.tag.copyBytes(to: cipherText + Int(plainTextSize), count: sealedBox.tag.count)
         } catch {
-            return AesGcmErrorCodes.EncryptionFailed
+            return ErrorCodes.encryptionFailed
         }
-        return AesGcmErrorCodes.Success
+        return ErrorCodes.success
     }
 }
+
+@objc(WebCryptoAesKwReturnValue)
+public final class AesKwReturnValue: NSObject {
+    @objc public var errCode: ErrorCodes = ErrorCodes.success
+    @objc public var outputSize: UInt64 = 0
+}
+
+@objc(WebCryptoAesKw)
+public final class AesKw: NSObject {
+    @objc public static func wrap(
+        _ key: UnsafePointer<UInt8>,
+        keySize: UInt,
+        data: UnsafePointer<UInt8>,
+        dataSize: UInt,
+        cipherText: UnsafeMutablePointer<UInt8>,
+        cipherTextSize: UInt64
+    ) -> AesKwReturnValue {
+        let rv = AesKwReturnValue()
+        rv.errCode = ErrorCodes.success
+        if keySize > Int.max
+            || dataSize > Int.max
+            || cipherTextSize > Int.max
+            || keySize == 0
+            || dataSize == 0
+            || cipherTextSize == 0
+        {
+            rv.errCode = ErrorCodes.invalidArgument
+            return rv
+        }
+        do {
+            let cipher = try AES.KeyWrap.wrap(
+                SymmetricKey(data: UnsafeBufferPointer(start: data, count: Int(dataSize))),
+                using: SymmetricKey(data: UnsafeBufferPointer(start: key, count: Int(keySize))))
+            if cipher.count > Int(cipherTextSize) {
+                rv.errCode = ErrorCodes.encryptionFailed
+                return rv
+            }
+            cipher.copyBytes(to: cipherText, count: cipher.count)
+            rv.errCode = ErrorCodes.success
```

Comment:
Does it work to just have "rv.errCode = .success"?

Reply:
yes. Changing